### ISSUE TITLE
Installation issue: DCMTK with TIFF support

### ISSUE THREAD

**Issue by moloney** [LLNL/spack#45198]
### Steps to reproduce

Trying to build dcmtk with tiff support fails during the linking stage.  While asking for help in the slack channel I was informed that the issue is the libtiff recipe needs to be updated to include a `libs()` method simliar to libpng.

### Error message

Building dcmtk with tiff support should work.

### Information on your system

* **Spack:** 0.23.0.dev0 (e5f5749d676d0cc4e247dd41c73bc31ff6b6945e)
* **Python:** 3.9.17
* **Platform:** linux-ubuntu16.04-ivybridge
* **Concretizer:** clingo

### General information

- [X] I have run `spack debug report` and reported the version of Spack/Python/Platform
- [X] I have searched the issues of this repo and believe this is not a duplicate
- [X] I have run the failing commands in debug mode and reported the output

**Comment by moloney** [comment]
@adamjstewart You mentioned in Slack to assign you, which I don't have the capability to do, so just giving a mention.

**Comment by adamjstewart** [comment]
@moloney can you give me the output of:
```console
> ls <prefix>/lib
```
You can get `<prefix>` from `spack find -p libtiff`. I just want to see how the libraries are named for you. For me on macOS, they are `libtiff.dylib`, which should be found by the default `libs()` method.

**Comment by moloney** [comment]
Here is the output:

cmake
libtiff.so
libtiff.so.6
libtiff.so.6.0.2
libtiffxx.so
libtiffxx.so.6
libtiffxx.so.6.0.2
pkgconfig

**Comment by adamjstewart** [comment]
Seems like it should work. Remind me what error you were seeing?

**Comment by moloney** [comment]
My original issue was building `dcmtk` with tiff support, in Slack it was suggested the issue was due to the missing `libs()` method.

The error when building dcmtk is:

```
[ 71%] Linking CXX executable ../../bin/dcm2pnm
cd /bigtmp/tmp/moloney/spack-stage/spack-stage-dcmtk-3.6.7-vnlggacdmss5wdx4hy4wndka2eapd3fw/spack-build-vnlggac/dcmimage/apps && /usr/contrib/byoe/repo/._internal/spack/opt/spack/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder/linux-ubuntu16.04-ivybridge/gcc-12.3.0/cmake-3.29.4-di2xxf45fr624obgrzeqbymb7eusi6ym/bin/cmake -E cmake_link_script CMakeFiles/dcm2pnm.dir/link.txt --verbose=1
/usr/contrib/byoe/repo/._internal/spack/lib/spack/env/gcc/g++  -D_XOPEN_SOURCE_EXTENDED -D_XOPEN_SOURCE=500 -D_BSD_SOURCE -D_DEFAULT_SOURCE -D_BSD_COMPAT -D_OSF_SOURCE -D_POSIX_C_SOURCE=199506L -O3 -DNDEBUG CMakeFiles/dcm2pnm.dir/dcm2pnm.cc.o -o ../../bin/dcm2pnm  -Wl,-rpath,/usr/contrib/byoe/repo/._internal/spack/opt/spack/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder/linux-ubuntu16.04-ivybridge/gcc-12.3.0/zlib-ng-2.1.6-uqbferhpitbgl3cgko4gcq2vly7wmjrk/lib:/usr/contrib/byoe/repo/._internal/spack/opt/spack/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder/linux-ubuntu16.04-ivybridge/gcc-12.3.0/libxml2-2.10.3-kb6x5eijzqczhjonnipzgkoui3ss5u5k/lib:/usr/contrib/byoe/repo/._internal/spack/opt/spack/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder/linux-ubuntu16.04-ivybridge/gcc-12.3.0/libiconv-1.17-fmk4tnaxichiitzhpladhukugxxjvpjv/lib:/usr/contrib/byoe/repo/._internal/spack/opt/spack/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder/linux-ubuntu16.04-ivybridge/gcc-12.3.0/libjpeg-turbo-3.0.0-6fvwoez4sgareo4f7wc3c6vasbwgjrhl/lib:/usr/contrib/byoe/repo/._internal/spack/opt/spack/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder/linux-ubuntu16.04-ivybridge/gcc-12.3.0/libpng-1.6.39-klgrztcrtd643w7jmxayhieta5w66hei/lib:::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::: ../../lib/libdcmimage.a ../../lib/libdcmimgle.a ../../lib/libdcmdata.a ../../lib/liboflog.a ../../lib/libofstd.a /usr/contrib/byoe/repo/._internal/spack/opt/spack/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder/linux-ubuntu16.04-ivybridge/gcc-12.3.0/zlib-ng-2.1.6-uqbferhpitbgl3cgko4gcq2vly7wmjrk/lib/libz.so /usr/contrib/byoe/repo/._internal/spack/opt/spack/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder/linux-ubuntu16.04-ivybridge/gcc-12.3.0/libxml2-2.10.3-kb6x5eijzqczhjonnipzgkoui3ss5u5k/lib/libxml2.so /usr/contrib/byoe/repo/._internal/spack/opt/spack/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder/linux-ubuntu16.04-ivybridge/gcc-12.3.0/libiconv-1.17-fmk4tnaxichiitzhpladhukugxxjvpjv/lib/libiconv.so /usr/contrib/byoe/repo/._internal/spack/opt/spack/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder/linux-ubuntu16.04-ivybridge/gcc-12.3.0/libiconv-1.17-fmk4tnaxichiitzhpladhukugxxjvpjv/lib/libcharset.so -lnsl -lpthread /usr/contrib/byoe/repo/._internal/spack/opt/spack/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder/linux-ubuntu16.04-ivybridge/gcc-12.3.0/libjpeg-turbo-3.0.0-6fvwoez4sgareo4f7wc3c6vasbwgjrhl/lib/libjpeg.so /usr/contrib/byoe/repo/._internal/spack/opt/spack/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder/linux-ubuntu16.04-ivybridge/gcc-12.3.0/libpng-1.6.39-klgrztcrtd643w7jmxayhieta5w66hei/lib/libpng.so
/usr/contrib/byoe/repo/._internal/spack/opt/spack/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder/linux-ubuntu16.04-ivybridge/gcc-5.4.0/binutils-2.41-fxgfdiqsjh2bp3y5enqq4nzberrctbc3/bin/ld: ../../lib/libdcmimage.a(dipitiff.cc.o): in function `DiTIFFPlugin::write(DiImage*, _IO_FILE*, unsigned long) const':
dipitiff.cc:(.text+0x179): undefined reference to `TIFFFdOpen'
/usr/contrib/byoe/repo/._internal/spack/opt/spack/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder/linux-ubuntu16.04-ivybridge/gcc-5.4.0/binutils-2.41-fxgfdiqsjh2bp3y5enqq4nzberrctbc3/bin/ld: dipitiff.cc:(.text+0x198): undefined reference to `TIFFSetField'
/usr/contrib/byoe/repo/._internal/spack/opt/spack/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder/linux-ubuntu16.04-ivybridge/gcc-5.4.0/binutils-2.41-fxgfdiqsjh2bp3y5enqq4nzberrctbc3/bin/ld: dipitiff.cc:(.text+0x1ab): undefined reference to `TIFFSetField'
/usr/contrib/byoe/repo/._internal/spack/opt/spack/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder/linux-ubuntu16.04-ivybridge/gcc-5.4.0/binutils-2.41-fxgfdiqsjh2bp3y5enqq4nzberrctbc3/bin/ld: dipitiff.cc:(.text+0x1bf): undefined reference to `TIFFSetField'
/usr/contrib/byoe/repo/._internal/spack/opt/spack/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder/linux-ubuntu16.04-ivybridge/gcc-5.4.0/binutils-2.41-fxgfdiqsjh2bp3y5enqq4nzberrctbc3/bin/ld: dipitiff.cc:(.text+0x1d3): undefined reference to `TIFFSetField'
/usr/contrib/byoe/repo/._internal/spack/opt/spack/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder/linux-ubuntu16.04-ivybridge/gcc-5.4.0/binutils-2.41-fxgfdiqsjh2bp3y5enqq4nzberrctbc3/bin/ld: dipitiff.cc:(.text+0x1e6): undefined reference to `TIFFSetField'
/usr/contrib/byoe/repo/._internal/spack/opt/spack/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder/linux-ubuntu16.04-ivybridge/gcc-5.4.0/binutils-2.41-fxgfdiqsjh2bp3y5enqq4nzberrctbc3/bin/ld: ../../lib/libdcmimage.a(dipitiff.cc.o):dipitiff.cc:(.text+0x206): more undefined references to `TIFFSetField' follow
/usr/contrib/byoe/repo/._internal/spack/opt/spack/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder/linux-ubuntu16.04-ivybridge/gcc-5.4.0/binutils-2.41-fxgfdiqsjh2bp3y5enqq4nzberrctbc3/bin/ld: ../../lib/libdcmimage.a(dipitiff.cc.o): in function `DiTIFFPlugin::write(DiImage*, _IO_FILE*, unsigned long) const':
dipitiff.cc:(.text+0x2b7): undefined reference to `TIFFWriteScanline'
/usr/contrib/byoe/repo/._internal/spack/opt/spack/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder/linux-ubuntu16.04-ivybridge/gcc-5.4.0/binutils-2.41-fxgfdiqsjh2bp3y5enqq4nzberrctbc3/bin/ld: dipitiff.cc:(.text+0x2da): undefined reference to `TIFFFlushData'
/usr/contrib/byoe/repo/._internal/spack/opt/spack/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder/linux-ubuntu16.04-ivybridge/gcc-5.4.0/binutils-2.41-fxgfdiqsjh2bp3y5enqq4nzberrctbc3/bin/ld: dipitiff.cc:(.text+0x2e2): undefined reference to `TIFFCleanup'
/usr/contrib/byoe/repo/._internal/spack/opt/spack/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder/linux-ubuntu16.04-ivybridge/gcc-5.4.0/binutils-2.41-fxgfdiqsjh2bp3y5enqq4nzberrctbc3/bin/ld: dipitiff.cc:(.text+0x356): undefined reference to `TIFFSetField'
/usr/contrib/byoe/repo/._internal/spack/opt/spack/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder/linux-ubuntu16.04-ivybridge/gcc-5.4.0/binutils-2.41-fxgfdiqsjh2bp3y5enqq4nzberrctbc3/bin/ld: ../../lib/libdcmimage.a(dipitiff.cc.o): in function `DiTIFFPlugin::getLibraryVersionString[abi:cxx11]()':
dipitiff.cc:(.text+0x3e2): undefined reference to `TIFFGetVersion'
collect2: error: ld returned 1 exit status
make[2]: *** [dcmimage/apps/CMakeFiles/dcm2pnm.dir/build.make:111: bin/dcm2pnm] Error 1
```

**Comment by adamjstewart** [comment]
Fixed by #45213

**Comment by adamjstewart** [comment]
Note that no `libs()` method is needed, the default one works fine. The bug was in DCMTK's CMake, not in Spack.
